app/component/itinerary/ItineraryPageUtils.js

```diff
@@ -535,3 +536,52 @@ export function quitIteration(plan, newPlan, planParams, startTime) {
   }
   return false;
 }
+
+/**
+ * Checks if itinerary in local storage was set with identical URL parameters.
+ *
+ * Note: storedParams.hash is not compared
+ *
+ * @param {{from: string, to: string, time: number, arriveBy: boolean}} storedParams
+ * @param {matchShape} match matchContext with URL params
+ * @returns true if stored itinerary was set with identical parameters
+ */
+const storedParamsMatchURL = (storedParams, match) => {
+  if (!storedParams || !match) {
+    return false;
+  }
+
+  return (
+    storedParams.from === match.params?.from &&
+    storedParams.to === match.params?.to &&
+    storedParams.time === match.location?.query?.time &&
+    storedParams.arriveBy === match.location?.query?.arriveBy
+  );
+};
+
+/**
+ * Enables Navigator assisted journey on itinerary selection if all of the following resolve as true:
+ * - a stored itinerary exists
+ * - the stored itinerary ends in future
+ * - the params stored along itinerary match those of the current query
+ *
+ * Note: Itinerary related hash param is also compared
+ *
+ * @param {matchShape} match matchContext with URL params
+ * @returns true if Navigator can be initialized with stored itinerary
+ */
+export const isStoredItineraryRelevant = match => {
+  /* eslint-disable eqeqeq */
+  const { itinerary, params } = getLatestNavigatorItinerary();
+  const {
+    params: { hash, secondHash },
+  } = match;
+
+  return (
+    (itinerary &&
+      Date.parse(itinerary.end) > Date.now() &&
+      storedParamsMatchURL(params, match) &&
+      params?.index == secondHash) ||
+    params?.index == hash
+  );
+};
```

Comment:
- last condition is apparently wrong, shoudl be ( params.index == secondHash || params.index == hash )
- would it be easier and safer to save hash and secondHash and compare both ?
- let code crash if match params is not defined. unnecessary validation to keep code running  although everyhing is all wrong is not wise